client/src/components/games/AvailableGames.test.tsx

```diff
@@ -0,0 +1,153 @@
+import { vi, describe, expect, it} from "vitest"
+import { render, screen } from "@testing-library/react"
+import userEvent from "@testing-library/user-event"
+import { BrowserRouter, Route, Routes } from "react-router-dom";
+import { Game } from "../../@types/game";
+import { Venue } from "../../@types/venue";
+import { GamesContext, VenuesContext } from "../../App";
+import AvailableGames from "./AvailableGames";
+import { timeDisplay } from "../../utilities/date-time-display";
+
+const mockVenues: Venue[] = [
+    {
+    _id: "1",
+    name: "Stadium",
+    address: "1234 Main St",
+    capacity: 10,
+    image: "https://example.com/image.jpg"
+  },
+  {
+    _id: "2",
+    name: "Arena",
+    address: "1 Arena St",
+    capacity: 20,
+    image: "https://example.com/image.jpg"
+  }
+]
+
+const pastDate = new Date(Date.now()-100000).toISOString();
+const nearFutureDate = new Date(Date.now()+100000).toISOString();
+const farFutureDate = new Date(Date.now()+10000000).toISOString();
+
+const mockGames: Game[] = [
+  {
+    _id: "1",
+    venue: mockVenues[0],
+    date: pastDate,
+    number_of_players_needed: 1,
+    players: [],
+    game_type: "5-a-side",
+    duration: 90,
+    price_per_head: 5,
+    contact_details: "123456789"
+  },
+  {
+    _id: "2",
+    venue: mockVenues[0],
+    date: nearFutureDate,
+    number_of_players_needed: 1,
+    players: [],
+    game_type: "8-a-side",
+    duration: 90,
+    price_per_head: 5,
+    contact_details: "123456789"
+  },
+  {
+    _id: "3",
+    venue: mockVenues[1],
+    date: farFutureDate,
+    number_of_players_needed: 1,
+    players: [],
+    game_type: "11-a-side",
+    duration: 90,
+    price_per_head: 5,
+    contact_details: "123456789"
+  }
+]
+
+describe("AvailableGames", () => {
+
+  it("has the same number of .availableGame as games provided", async () => {
+    const { container } = render(
+      <BrowserRouter>
```

Comment:
I think its better to use MemoryRouter here. It simulates navigating without affecting the URL in the browser. I believe not really important here, but still